src/advanced_inv.cpp

```diff
@@ -913,14 +913,24 @@ bool advanced_inventory::move_all_items()
 
     // Check some preconditions to quickly leave the function.
     size_t liquid_items = 0;
+    size_t gas_items = 0;
     for( const advanced_inv_listitem &elem : spane.items ) {
         for( const item_location &elemit : elem.items ) {
             if( elemit->made_of_from_type( phase_id::LIQUID ) && !elemit->is_frozen_liquid() ) {
                 liquid_items++;
             }
         }
     }
-    if( spane.items.empty() || liquid_items == spane.items.size() ) {
+
+    for( const advanced_inv_listitem &elem : spane.items ) {
+        for( const item_location &elemit : elem.items ) {
+            if( elemit->made_of_from_type( phase_id::GAS ) ) {
+                gas_items++;
+            }
+        }
+    }
+    if( spane.items.empty() || liquid_items == spane.items.size() || gas_items == spane.items.size() ) {
```

Comment:
Kinda like the other condition, unless you explicitly need the count of liquids and gases, count the solids and return if it's 0. That also lets you exit the loop early. And you can use `item::made_of` instead of `item::made_of_from_type` if you want frozen items to be handleable.

If you do need them explicitly, count them in the same loop. And I guess compare the size to liquids+gases in the condition instead of both individually?